Importamos las librerias que nos hacen falta

In [1]:
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt 
import seaborn as sns
from collections import Counter
from scipy.spatial import distance     

Vamos a crear un dataset con el archivo csv 'movies-dataset'

In [2]:
df_peliculas = pd.read_csv ('..\Proyecto individual Henry\Data\movies_dataset (1).csv')


C:\Users\Andrea\AppData\Local\Temp\ipykernel_13920\270986259.py:1: DtypeWarning: Columns (10) have mixed types. Specify dtype option on import or set low_memory=False.
  df_peliculas = pd.read_csv ('..\Proyecto individual Henry\Data\movies_dataset (1).csv')


Vemos el contenido del dataset

In [3]:
df_peliculas.head(15)

adult                              belongs_to_collection    budget  \
0   False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1   False                                                NaN  65000000   
2   False  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3   False                                                NaN  16000000   
4   False  {'id': 96871, 'name': 'Father of the Bride Col...         0   
5   False                                                NaN  60000000   
6   False                                                NaN  58000000   
7   False                                                NaN         0   
8   False                                                NaN  35000000   
9   False  {'id': 645, 'name': 'James Bond Collection', '...  58000000   
10  False                                                NaN  62000000   
11  False                                                NaN         0   
12  False  {'id': 117693, 'name': 'Balto Collection', 'po...         0   
13  False                                                NaN  44000000   
14  False                                                NaN  98000000   

                                               genres  \
0   [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1   [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2   [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3   [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4                      [{'id': 35, 'name': 'Comedy'}]   
5   [{'id': 28, 'name': 'Action'}, {'id': 80, 'nam...   
6   [{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...   
7   [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
8   [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
9   [{'id': 12, 'name': 'Adventure'}, {'id': 28, '...   
10  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
11  [{'id': 35, 'name': 'Comedy'}, {'id': 27, 'nam...   
12  [{'id': 10751, 'name': 'Family'}, {'id': 16, '...   
13  [{'id': 36, 'name': 'History'}, {'id': 18, 'na...   
14  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   

                                        homepage     id    imdb_id  \
0           http://toystory.disney.com/toy-story    862  tt0114709   
1                                            NaN   8844  tt0113497   
2                                            NaN  15602  tt0113228   
3                                            NaN  31357  tt0114885   
4                                            NaN  11862  tt0113041   
5                                            NaN    949  tt0113277   
6                                            NaN  11860  tt0114319   
7                                            NaN  45325  tt0112302   
8                                            NaN   9091  tt0114576   
9   http://www.mgm.com/view/movie/757/Goldeneye/    710  tt0113189   
10                                           NaN   9087  tt0112346   
11                                           NaN  12110  tt0112896   
12                                           NaN  21032  tt0112453   
13                                           NaN  10858  tt0113987   
14                                           NaN   1408  tt0112760   

   original_language               original_title  \
0                 en                    Toy Story   
1                 en                      Jumanji   
2                 en             Grumpier Old Men   
3                 en            Waiting to Exhale   
4                 en  Father of the Bride Part II   
5                 en                         Heat   
6                 en                      Sabrina   
7                 en                 Tom and Huck   
8                 en                 Sudden Death   
9                 en                    GoldenEye   
10                en       The American President   
11                en  Dracula: Dead and Loving It   
12                en                        Balto   
13                en                       

Vamos a ver como la información del df_peliculas

In [4]:
df_peliculas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 24 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   adult                  45466 non-null  object 
 1   belongs_to_collection  4494 non-null   object 
 2   budget                 45466 non-null  object 
 3   genres                 45466 non-null  object 
 4   homepage               7782 non-null   object 
 5   id                     45466 non-null  object 
 6   imdb_id                45449 non-null  object 
 7   original_language      45455 non-null  object 
 8   original_title         45466 non-null  object 
 9   overview               44512 non-null  object 
 10  popularity             45461 non-null  object 
 11  poster_path            45080 non-null  object 
 12  production_companies   45463 non-null  object 
 13  production_countries   45463 non-null  object 
 14  release_date           45379 non-null  object 
 15  re

Vamos a rellenar los nulos de las columna revenue por cero.

In [5]:
df_peliculas['revenue'].fillna(0)

0        373554033.0
1        262797249.0
2                0.0
3         81452156.0
4         76578911.0
            ...     
45461            0.0
45462            0.0
45463            0.0
45464            0.0
45465            0.0
Name: revenue, Length: 45466, dtype: float64

Vamos a rellenar los nulos de las columna budget por cero.

In [6]:
df_peliculas['budget'].replace('NaN',0)

0        30000000
1        65000000
2               0
3        16000000
4               0
           ...   
45461           0
45462           0
45463           0
45464           0
45465           0
Name: budget, Length: 45466, dtype: object

Eliminamos valores nulos de la columna release_date

In [7]:
df_peliculas = df_peliculas.dropna(subset=['release_date'])

Vemos la columna release_date del df_peliculas

In [8]:
df_peliculas['release_date']

0        1995-10-30
1        1995-12-15
2        1995-12-22
3        1995-12-22
4        1995-02-10
            ...    
45460    1991-05-13
45462    2011-11-17
45463    2003-08-01
45464    1917-10-21
45465    2017-06-09
Name: release_date, Length: 45379, dtype: object

Agregamos el año de la fecha de extreno en una columna release_year

In [9]:
df_peliculas['release_year']= df_peliculas['release_date'].str.slice(0,4)

Vemos la columna release_year 

In [10]:
df_peliculas['release_year']

0        1995
1        1995
2        1995
3        1995
4        1995
         ... 
45460    1991
45462    2011
45463    2003
45464    1917
45465    2017
Name: release_year, Length: 45379, dtype: object

Vemos la informacion del dataset df_peliculas

In [11]:
df_peliculas.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 45379 entries, 0 to 45465
Data columns (total 25 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   adult                  45379 non-null  object 
 1   belongs_to_collection  4491 non-null   object 
 2   budget                 45379 non-null  object 
 3   genres                 45379 non-null  object 
 4   homepage               7769 non-null   object 
 5   id                     45379 non-null  object 
 6   imdb_id                45365 non-null  object 
 7   original_language      45368 non-null  object 
 8   original_title         45379 non-null  object 
 9   overview               44438 non-null  object 
 10  popularity             45377 non-null  object 
 11  poster_path            45040 non-null  object 
 12  production_companies   45379 non-null  object 
 13  production_countries   45379 non-null  object 
 14  release_date           45379 non-null  object 
 15  re

Importamos datetime de datetime

In [12]:
from datetime import datetime

Convertimos la columna release_date en tipo de dato fecha

In [13]:
df_peliculas['release_date']= pd.to_datetime(df_peliculas['release_date'], errors='coerce')

Volvemos a ver la informacion df_peliculas donde figura datetime como tipo de dato de la columna release_date

In [14]:
df_peliculas.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 45379 entries, 0 to 45465
Data columns (total 25 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   adult                  45379 non-null  object        
 1   belongs_to_collection  4491 non-null   object        
 2   budget                 45379 non-null  object        
 3   genres                 45379 non-null  object        
 4   homepage               7769 non-null   object        
 5   id                     45379 non-null  object        
 6   imdb_id                45365 non-null  object        
 7   original_language      45368 non-null  object        
 8   original_title         45379 non-null  object        
 9   overview               44438 non-null  object        
 10  popularity             45377 non-null  object        
 11  poster_path            45040 non-null  object        
 12  production_companies   45379 non-null  object        
 13  p

Cambiamos los nulos por cero de la columna release_year

In [15]:
df_peliculas['release_year']= df_peliculas['release_year'].replace('NaN', 0)
df_peliculas['release_year']

0        1995
1        1995
2        1995
3        1995
4        1995
         ... 
45460    1991
45462    2011
45463    2003
45464    1917
45465    2017
Name: release_year, Length: 45379, dtype: object

Vemos los últimos 15 datos

In [16]:
df_peliculas.head(-15)

adult                              belongs_to_collection    budget  \
0      False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1      False                                                NaN  65000000   
2      False  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3      False                                                NaN  16000000   
4      False  {'id': 96871, 'name': 'Father of the Bride Col...         0   
...      ...                                                ...       ...   
45445  False                                                NaN         0   
45446  False                                                NaN         0   
45447  False                                                NaN         0   
45448  False                                                NaN         0   
45449  False                                                NaN         0   

                                                  genres  \
0      [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1      [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2      [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3      [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4                         [{'id': 35, 'name': 'Comedy'}]   
...                                                  ...   
45445  [{'id': 14, 'name': 'Fantasy'}, {'id': 35, 'na...   
45446  [{'id': 14, 'name': 'Fantasy'}, {'id': 35, 'na...   
45447                                                 []   
45448                                                 []   
45449  [{'id': 16, 'name': 'Animation'}, {'id': 10751...   

                                   homepage      id    imdb_id  \
0      http://toystory.disney.com/toy-story     862  tt0114709   
1                                       NaN    8844  tt0113497   
2                                       NaN   15602  tt0113228   
3                                       NaN   31357  tt0114885   
4                                       NaN   11862  tt0113041   
...                                     ...     ...        ...   
45445                                   NaN   49277  tt0135122   
45446                                   NaN   49271  tt0127948   
45447                                   NaN   44324  tt0135631   
45448                                   NaN  122036  tt0224286   
45449                                   NaN   14885  tt0457437   

      original_language                     original_title  \
0                    en                          Toy Story   
1                    en                            Jumanji   
2                    en                   Grumpier Old Men   
3                    en                  Waiting to Exhale   
4                    en        Father of the Bride Part II   
...                 ...                                ...   
45445                en           Les affiches en goguette   
45446                en            Le locataire diabolique   
45447                fr               Le Roi du maquillage   
45448                fr  Les Transmutations imperceptibles   
45449                en   Pooh's Heffalump Halloween Movie   

                                                overview  ...      revenue  \
0      Led by Woody, Andy's toys live happily in his ...  ...  373554033.0   
1      When siblings Judy and Peter discover an encha...  ...  262797249.0   
2      A family wedding reignites the ancient feud be...  ...          0.0   
3      Cheated on, mistreated and stepped on, the wom...  ...   81452156.0   
4      Just when George Banks has recovered from his ...  ...   76578911.0   
...                                                  ...  ...          ...   
45445  A wall full of advertising posters comes to life.  ...          0.0   
45446  A man rents an apartment and furnishes it in r...  ...          0.0   
45447  The background of this picture represents a sc...  ...          0.0   
45448  This shows a prince entering upon th

Eliminamos nulos de la columna release_year

In [17]:
df_peliculas=df_peliculas.dropna(subset=['release_year'])

Vemos los tipos de datos del dataset

In [18]:
df_peliculas.dtypes

adult                            object
belongs_to_collection            object
budget                           object
genres                           object
homepage                         object
id                               object
imdb_id                          object
original_language                object
original_title                   object
overview                         object
popularity                       object
poster_path                      object
production_companies             object
production_countries             object
release_date             datetime64[ns]
revenue                         float64
runtime                         float64
spoken_languages                 object
status                           object
tagline                          object
title                            object
video                            object
vote_average                    float64
vote_count                      float64
release_year                     object


Cambiar los valores no finitos de la columna release_year por cero

In [19]:
df_peliculas['release_year']= df_peliculas['release_year'].fillna(0)

In [20]:
df_peliculas['release_year']

0        1995
1        1995
2        1995
3        1995
4        1995
         ... 
45460    1991
45462    2011
45463    2003
45464    1917
45465    2017
Name: release_year, Length: 45379, dtype: object

Cambio el tipo de dato de la columna release_year de float a int 

In [21]:
df_peliculas['release_year']=df_peliculas['release_year'].astype(int)

Eliminamos nulos de la columna 'title'

In [22]:
df_peliculas= df_peliculas.dropna(subset=['title'])

Vemos los 3 primeros datos del dataset

In [23]:
df_peliculas.head(3)

adult                              belongs_to_collection    budget  \
0  False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1  False                                                NaN  65000000   
2  False  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   

                                              genres  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   

                               homepage     id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story    862  tt0114709                en   
1                                   NaN   8844  tt0113497                en   
2                                   NaN  15602  tt0113228                en   

     original_title                                           overview  ...  \
0         Toy Story  Led by Woody, Andy's toys live happily in his ...  ...   
1           Jumanji  When siblings Judy and Peter discover an encha...  ...   
2  Grumpier Old Men  A family wedding reignites the ancient feud be...  ...   

       revenue runtime                                   spoken_languages  \
0  373554033.0    81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0   104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0   101.0           [{'iso_639_1': 'en', 'name': 'English'}]   

     status                                            tagline  \
0  Released                                                NaN   
1  Released          Roll the dice and unleash the excitement!   
2  Released  Still Yelling. Still Fighting. Still Ready for...   

              title  video vote_average vote_count release_year  
0         Toy Story  False          7.7     5415.0         1995  
1           Jumanji  False          6.9     2413.0         1995  
2  Grumpier Old Men  False          6.5       92.0         1995  

[3 rows x 25 columns]

Volvemos a ver la información

In [24]:
df_peliculas.head(5)

adult                              belongs_to_collection    budget  \
0  False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1  False                                                NaN  65000000   
2  False  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3  False                                                NaN  16000000   
4  False  {'id': 96871, 'name': 'Father of the Bride Col...         0   

                                              genres  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4                     [{'id': 35, 'name': 'Comedy'}]   

                               homepage     id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story    862  tt0114709                en   
1                                   NaN   8844  tt0113497                en   
2                                   NaN  15602  tt0113228                en   
3                                   NaN  31357  tt0114885                en   
4                                   NaN  11862  tt0113041                en   

                original_title  \
0                    Toy Story   
1                      Jumanji   
2             Grumpier Old Men   
3            Waiting to Exhale   
4  Father of the Bride Part II   

                                            overview  ...      revenue  \
0  Led by Woody, Andy's toys live happily in his ...  ...  373554033.0   
1  When siblings Judy and Peter discover an encha...  ...  262797249.0   
2  A family wedding reignites the ancient feud be...  ...          0.0   
3  Cheated on, mistreated and stepped on, the wom...  ...   81452156.0   
4  Just when George Banks has recovered from his ...  ...   76578911.0   

  runtime                                   spoken_languages    status  \
0    81.0           [{'iso_639_1': 'en', 'name': 'English'}]  Released   
1   104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...  Released   
2   101.0           [{'iso_639_1': 'en', 'name': 'English'}]  Released   
3   127.0           [{'iso_639_1': 'en', 'name': 'English'}]  Released   
4   106.0           [{'iso_639_1': 'en', 'name': 'English'}]  Released   

                                             tagline  \
0                                                NaN   
1          Roll the dice and unleash the excitement!   
2  Still Yelling. Still Fighting. Still Ready for...   
3  Friends are the people who let you be yourself...   
4  Just When His World Is Back To Normal... He's ...   

                         title  video vote_average vote_count release_year  
0                    Toy Story  False          7.7     5415.0         1995  
1                      Jumanji  False          6.9     2413.0         1995  
2             Grumpier Old Men  False          6.5       92.0         1995  
3            Waiting to Exhale  False          6.1       34.0         1995  
4  Father of the Bride Part II  False          5.7      173.0         1995  

[5 rows x 25 columns]

Vemos los duplicados

In [25]:
new_duplicate= df_peliculas[df_peliculas.duplicated('id')]
print(new_duplicate)

       adult                              belongs_to_collection    budget  \
1465   False                                                NaN         0   
9165   False                                                NaN         0   
9327   False                                                NaN         0   
12066  False                                                NaN   1600000   
13375  False                                                NaN         0   
15074  False                                                NaN         4   
15765  False                                                NaN      2500   
16764  False                                                NaN         0   
20843  False                                                NaN  40000000   
20899  False                                                NaN         0   
21116  False                                                NaN         0   
21165  False                                                NaN         0   

Eliminamos los duplicados teniendo en cuenta la columna id

In [26]:
df_peliculas.drop_duplicates(subset=['id'], inplace=True)

Vemos la informacion del df_peliculas nuevamente

In [27]:
df_peliculas.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 45346 entries, 0 to 45465
Data columns (total 25 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   adult                  45346 non-null  object        
 1   belongs_to_collection  4485 non-null   object        
 2   budget                 45346 non-null  object        
 3   genres                 45346 non-null  object        
 4   homepage               7761 non-null   object        
 5   id                     45346 non-null  object        
 6   imdb_id                45332 non-null  object        
 7   original_language      45335 non-null  object        
 8   original_title         45346 non-null  object        
 9   overview               44405 non-null  object        
 10  popularity             45346 non-null  object        
 11  poster_path            45007 non-null  object        
 12  production_companies   45346 non-null  object        
 13  p

Vemos solo los tipos de la columnas

In [28]:
df_peliculas.dtypes

adult                            object
belongs_to_collection            object
budget                           object
genres                           object
homepage                         object
id                               object
imdb_id                          object
original_language                object
original_title                   object
overview                         object
popularity                       object
poster_path                      object
production_companies             object
production_countries             object
release_date             datetime64[ns]
revenue                         float64
runtime                         float64
spoken_languages                 object
status                           object
tagline                          object
title                            object
video                            object
vote_average                    float64
vote_count                      float64
release_year                      int32


Vamos a cambiar el tipo de dato de la columna budget de objet a float.

In [29]:
df_peliculas['budget']= pd.to_numeric(df_peliculas['budget'], errors='coerce')

Vamos a crear la columna return dividiendo la columna revenue en budget

In [30]:
df_peliculas['return'] = df_peliculas['revenue']/df_peliculas['budget']

En la columna return vamos a remplazar valores nulos por ceros

In [31]:
df_peliculas['return']= df_peliculas['return'].fillna(0)

In [32]:
df_peliculas['return']

0        12.451801
1         4.043035
2         0.000000
3         5.090760
4              inf
           ...    
45460     0.000000
45462     0.000000
45463     0.000000
45464     0.000000
45465     0.000000
Name: return, Length: 45346, dtype: float64

Creo un df_collection copiando la columan belongs_to_collection

In [33]:
df_collection = df_peliculas[['belongs_to_collection']].copy()



In [34]:
import json
from pandas import json_normalize

In [35]:
print(df_collection)

                                   belongs_to_collection
0      {'id': 10194, 'name': 'Toy Story Collection', ...
1                                                    NaN
2      {'id': 119050, 'name': 'Grumpy Old Men Collect...
3                                                    NaN
4      {'id': 96871, 'name': 'Father of the Bride Col...
...                                                  ...
45460                                                NaN
45462                                                NaN
45463                                                NaN
45464                                                NaN
45465                                                NaN

[45346 rows x 1 columns]


Eliminamos nulos de la columna belongs_to_collection

In [36]:
df_collection = df_collection.dropna(subset=['belongs_to_collection'])

Accedemos a los valores de los diccionarios. Vamos a crear una columna id en df_collection

In [37]:
df_collection['id']=df_collection['belongs_to_collection'].apply(lambda x : x[7:13])


Vamos a remplazar valor para limpiar la columna id

In [38]:
df_collection['id']= df_collection['id'].replace(',','', regex=True)
df_collection['id']= df_collection['id'].replace(' ','', regex=True)
df_collection['id']= df_collection['id'].replace("'","", regex=True)


Creamos una columna name en df_collection

In [39]:
df_collection['name']=df_collection['belongs_to_collection'].apply(lambda x: x[20:])

Limpiamos esa columna name del df_collection

In [40]:
df_collection['name']= df_collection['name'].str.split(',',expand=True)[0]
df_collection['name']= df_collection['name'].str.replace(': ','')
df_collection['name']= df_collection['name'].str.replace("''","'")

Creamos otra columna poster_path en df_collection

In [41]:
df_collection['poster_path']=df_collection['belongs_to_collection'].apply(lambda x: x[57:])

In [42]:
df_collection['poster_path']

0        h': '/7G9915LfUQ2lVfwMEEhDsn3kT4B.jpg', 'backd...
2        er_path': '/nLvUdqgPgm3F85NMCii9gVFUcet.jpg', ...
4        poster_path': '/nts4iOmNnq7GNicycMJ9pSAn204.jp...
9        ': '/HORpg5CSkmeQlAolx3bKMrKgfi.jpg', 'backdro...
12        '/w0ZgH6Lgxt2bQYnf1ss74UvYftm.jpg', 'backdrop...
                               ...                        
45355    path': '/2P0HNrYgKDvirV8RCdT1rBSJdbJ.jpg', 'ba...
45358    path': '/2P0HNrYgKDvirV8RCdT1rBSJdbJ.jpg', 'ba...
45369    path': '/2P0HNrYgKDvirV8RCdT1rBSJdbJ.jpg', 'ba...
45371          'poster_path': None, 'backdrop_path': None}
45382    th': '/yf9Eod9ANXyHTzDpAxz9ezgvlL4.jpg', 'back...
Name: poster_path, Length: 4485, dtype: object

Limpiamos esa columna poster_path

In [43]:
df_collection['poster_path']= df_collection['poster_path'].str.replace("'poster_path':'",'')

Elimino todo lo que esta antes del espacio

In [44]:
df_collection['poster_path']= df_collection['poster_path'].str.split(" ",expand=True)[1]

Elimino de la columna poster_path todo lo que esta despues de la coma

In [45]:
df_collection['poster_path']= df_collection['poster_path'].str.split("\,",expand=True)[0]

Creamos un columna backdrop_path en el df_collection

In [46]:
df_collection['backdrop_path']=df_collection['belongs_to_collection'].apply(lambda x: x[110:])


Elimino de la columna backdrop_path todo lo que esta antes del espacio.

In [47]:
df_collection['backdrop_path']= df_collection['backdrop_path'].str.split(" ",expand=True)[1]

Elimino de la columna backdrop_path la llave.

In [48]:
df_collection['backdrop_path']= df_collection['backdrop_path'].str.replace('}','')

C:\Users\Andrea\AppData\Local\Temp\ipykernel_13920\1380748080.py:1: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  df_collection['backdrop_path']= df_collection['backdrop_path'].str.replace('}','')


Vemos como quedo el df_collection

In [49]:
df_collection

belongs_to_collection      id  \
0      {'id': 10194, 'name': 'Toy Story Collection', ...   10194   
2      {'id': 119050, 'name': 'Grumpy Old Men Collect...  119050   
4      {'id': 96871, 'name': 'Father of the Bride Col...   96871   
9      {'id': 645, 'name': 'James Bond Collection', '...     645   
12     {'id': 117693, 'name': 'Balto Collection', 'po...  117693   
...                                                  ...     ...   
45355  {'id': 37261, 'name': 'The Carry On Collection...   37261   
45358  {'id': 37261, 'name': 'The Carry On Collection...   37261   
45369  {'id': 37261, 'name': 'The Carry On Collection...   37261   
45371  {'id': 477208, 'name': 'DC Super Hero Girls Co...  477208   
45382  {'id': 200641, 'name': 'Red Lotus Collection',...  200641   

                                   name                         poster_path  \
0                'Toy Story Collection'  '/7G9915LfUQ2lVfwMEEhDsn3kT4B.jpg'   
2           'Grumpy Old Men Collection'  '/nLvUdqgPgm3F85NMCii9gVFUcet.jpg'   
4      'Father of the Bride Collection'  '/nts4iOmNnq7GNicycMJ9pSAn204.jpg'   
9               'James Bond Collection'   '/HORpg5CSkmeQlAolx3bKMrKgfi.jpg'   
12                   'Balto Collection'  '/w0ZgH6Lgxt2bQYnf1ss74UvYftm.jpg'   
...                                 ...                                 ...   
45355         'The Carry On Collection'  '/2P0HNrYgKDvirV8RCdT1rBSJdbJ.jpg'   
45358         'The Carry On Collection'  '/2P0HNrYgKDvirV8RCdT1rBSJdbJ.jpg'   
45369         'The Carry On Collection'  '/2P0HNrYgKDvirV8RCdT1rBSJdbJ.jpg'   
45371  'DC Super Hero Girls Collection'                                None   
45382            'Red Lotus Collection'  '/yf9Eod9ANXyHTzDpAxz9ezgvlL4.jpg'   

                            backdrop_path  
0      '/9FBwqcd9IRruEDUrTdcaafOMKUq.jpg'  
2      '/hypTnLot2z8wpFS7qwsQHW1uV8u.jpg'  
4      '/7qwE57OVZmMJChBpLEbJEmzUydk.jpg'  
9      '/6VcVl48kNKvdXOZfJPdarlUGOsk.jpg'  
12     '/9VM5LiJV0bGb1st1KyHA3cVnO2G.jpg'  
...                                   ...  
45355  '/38tF1LJN7ULeZAuAfP7beaPMfcl.jpg'  
45358  '/38tF1LJN7ULeZAuAfP7beaPMfcl.jpg'  
45369  '/38tF1LJN7ULeZAuAfP7beaPMfcl.jpg'  
45371                                None  
45382  '/3fhHbLeO3DqdHvgHg5szs399eBb.jpg'  

[4485 rows x 5 columns]

Eliminamos la columna belongs_to_collection del df_collection

In [50]:
df_collection= df_collection.drop('belongs_to_collection', axis=1)

Mostramos nuevamente el df_collection

In [51]:
df_collection

id                              name  \
0       10194            'Toy Story Collection'   
2      119050       'Grumpy Old Men Collection'   
4       96871  'Father of the Bride Collection'   
9         645           'James Bond Collection'   
12     117693                'Balto Collection'   
...       ...                               ...   
45355   37261         'The Carry On Collection'   
45358   37261         'The Carry On Collection'   
45369   37261         'The Carry On Collection'   
45371  477208  'DC Super Hero Girls Collection'   
45382  200641            'Red Lotus Collection'   

                              poster_path                       backdrop_path  
0      '/7G9915LfUQ2lVfwMEEhDsn3kT4B.jpg'  '/9FBwqcd9IRruEDUrTdcaafOMKUq.jpg'  
2      '/nLvUdqgPgm3F85NMCii9gVFUcet.jpg'  '/hypTnLot2z8wpFS7qwsQHW1uV8u.jpg'  
4      '/nts4iOmNnq7GNicycMJ9pSAn204.jpg'  '/7qwE57OVZmMJChBpLEbJEmzUydk.jpg'  
9       '/HORpg5CSkmeQlAolx3bKMrKgfi.jpg'  '/6VcVl48kNKvdXOZfJPdarlUGOsk.jpg'  
12     '/w0ZgH6Lgxt2bQYnf1ss74UvYftm.jpg'  '/9VM5LiJV0bGb1st1KyHA3cVnO2G.jpg'  
...                                   ...                                 ...  
45355  '/2P0HNrYgKDvirV8RCdT1rBSJdbJ.jpg'  '/38tF1LJN7ULeZAuAfP7beaPMfcl.jpg'  
45358  '/2P0HNrYgKDvirV8RCdT1rBSJdbJ.jpg'  '/38tF1LJN7ULeZAuAfP7beaPMfcl.jpg'  
45369  '/2P0HNrYgKDvirV8RCdT1rBSJdbJ.jpg'  '/38tF1LJN7ULeZAuAfP7beaPMfcl.jpg'  
45371                                None                                None  
45382  '/yf9Eod9ANXyHTzDpAxz9ezgvlL4.jpg'  '/3fhHbLeO3DqdHvgHg5szs399eBb.jpg'  

[4485 rows x 4 columns]

In [52]:
df_collection.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4485 entries, 0 to 45382
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   id             4485 non-null   object
 1   name           4485 non-null   object
 2   poster_path    4485 non-null   object
 3   backdrop_path  3542 non-null   object
dtypes: object(4)
memory usage: 175.2+ KB


In [53]:
df_collection.to_csv('collection.csv', index = False)


Procedemos a eliminar columnas: 'adult','homepage','imdb_id','original_title','video' y 'poster_path')

In [54]:
df_peliculas=df_peliculas.drop('adult', axis=1)

In [55]:
df_peliculas=df_peliculas.drop('homepage', axis=1)
df_peliculas=df_peliculas.drop('imdb_id', axis=1)
df_peliculas=df_peliculas.drop('original_title', axis=1)
df_peliculas=df_peliculas.drop('video', axis=1)


In [56]:
df_peliculas=df_peliculas.drop('poster_path', axis=1)

In [57]:
df_peliculas

belongs_to_collection    budget  \
0      {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1                                                    NaN  65000000   
2      {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3                                                    NaN  16000000   
4      {'id': 96871, 'name': 'Father of the Bride Col...         0   
...                                                  ...       ...   
45460                                                NaN         0   
45462                                                NaN         0   
45463                                                NaN         0   
45464                                                NaN         0   
45465                                                NaN         0   

                                                  genres      id  \
0      [{'id': 16, 'name': 'Animation'}, {'id': 35, '...     862   
1      [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...    8844   
2      [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   15602   
3      [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   31357   
4                         [{'id': 35, 'name': 'Comedy'}]   11862   
...                                                  ...     ...   
45460  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...   30840   
45462                      [{'id': 18, 'name': 'Drama'}]  111109   
45463  [{'id': 28, 'name': 'Action'}, {'id': 18, 'nam...   67758   
45464                                                 []  227506   
45465                                                 []  461257   

      original_language                                           overview  \
0                    en  Led by Woody, Andy's toys live happily in his ...   
1                    en  When siblings Judy and Peter discover an encha...   
2                    en  A family wedding reignites the ancient feud be...   
3                    en  Cheated on, mistreated and stepped on, the wom...   
4                    en  Just when George Banks has recovered from his ...   
...                 ...                                                ...   
45460                en  Yet another version of the classic epic, with ...   
45462                tl  An artist struggles to finish his work while a...   
45463                en  When one of her hits goes wrong, a professiona...   
45464                en  In a small town live two brothers, one a minis...   
45465                en  50 years after decriminalisation of homosexual...   

      popularity                               production_companies  \
0      21.946943     [{'name': 'Pixar Animation Studios', 'id': 3}]   
1      17.015539  [{'name': 'TriStar Pictures', 'id': 559}, {'na...   
2        11.7129  [{'name': 'Warner Bros.', 'id': 6194}, {'name'...   
3       3.859495  [{'name': 'Twentieth Century Fox Film Corporat...   
4       8.387519  [{'name': 'Sandollar Productions', 'id': 5842}...   
...          ...                                                ...   
45460   5.683753  [{'name': 'Westdeutscher Rundfunk (WDR)', 'id'...   
45462   0.178241             [{'name': 'Sine Olivia', 'id': 19653}]   
45463   0.903007  [{'name': 'American World Pictures', 'id': 6165}]   
45464   0.003503               [{'name': 'Yermoliev', 'id': 88753}]   
45465   0.163015                                                 []   

                                    production_countries release_date  \
0      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-10-30   
1      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-15   
2      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
3      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
4      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-02-10   
...                                                  ...          ...   
45460  [{'iso_3166_1': 'CA', 'name': 'Canada'}, {'iso...   1991-05-13   
454

In [58]:
df_peliculas.info()



<class 'pandas.core.frame.DataFrame'>
Int64Index: 45346 entries, 0 to 45465
Data columns (total 20 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   belongs_to_collection  4485 non-null   object        
 1   budget                 45346 non-null  int64         
 2   genres                 45346 non-null  object        
 3   id                     45346 non-null  object        
 4   original_language      45335 non-null  object        
 5   overview               44405 non-null  object        
 6   popularity             45346 non-null  object        
 7   production_companies   45346 non-null  object        
 8   production_countries   45346 non-null  object        
 9   release_date           45346 non-null  datetime64[ns]
 10  revenue                45346 non-null  float64       
 11  runtime                45100 non-null  float64       
 12  spoken_languages       45346 non-null  object        
 13  s

Vamos a aplicar ast.literal_eval a la columna genres

In [59]:
import ast
df_peliculas['genres']= df_peliculas['genres'].apply(ast.literal_eval)

La columna genres en df_peliculas solo va a contener el nombre del genero

In [60]:
df_peliculas['genres']= df_peliculas['genres'].apply(lambda x :[d['name'] for d in x])

In [61]:
df_peliculas['genres']

0         [Animation, Comedy, Family]
1        [Adventure, Fantasy, Family]
2                   [Romance, Comedy]
3            [Comedy, Drama, Romance]
4                            [Comedy]
                     ...             
45460        [Drama, Action, Romance]
45462                         [Drama]
45463       [Action, Drama, Thriller]
45464                              []
45465                              []
Name: genres, Length: 45346, dtype: object

In [62]:
df_peliculas

belongs_to_collection    budget  \
0      {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1                                                    NaN  65000000   
2      {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3                                                    NaN  16000000   
4      {'id': 96871, 'name': 'Father of the Bride Col...         0   
...                                                  ...       ...   
45460                                                NaN         0   
45462                                                NaN         0   
45463                                                NaN         0   
45464                                                NaN         0   
45465                                                NaN         0   

                             genres      id original_language  \
0       [Animation, Comedy, Family]     862                en   
1      [Adventure, Fantasy, Family]    8844                en   
2                 [Romance, Comedy]   15602                en   
3          [Comedy, Drama, Romance]   31357                en   
4                          [Comedy]   11862                en   
...                             ...     ...               ...   
45460      [Drama, Action, Romance]   30840                en   
45462                       [Drama]  111109                tl   
45463     [Action, Drama, Thriller]   67758                en   
45464                            []  227506                en   
45465                            []  461257                en   

                                                overview popularity  \
0      Led by Woody, Andy's toys live happily in his ...  21.946943   
1      When siblings Judy and Peter discover an encha...  17.015539   
2      A family wedding reignites the ancient feud be...    11.7129   
3      Cheated on, mistreated and stepped on, the wom...   3.859495   
4      Just when George Banks has recovered from his ...   8.387519   
...                                                  ...        ...   
45460  Yet another version of the classic epic, with ...   5.683753   
45462  An artist struggles to finish his work while a...   0.178241   
45463  When one of her hits goes wrong, a professiona...   0.903007   
45464  In a small town live two brothers, one a minis...   0.003503   
45465  50 years after decriminalisation of homosexual...   0.163015   

                                    production_companies  \
0         [{'name': 'Pixar Animation Studios', 'id': 3}]   
1      [{'name': 'TriStar Pictures', 'id': 559}, {'na...   
2      [{'name': 'Warner Bros.', 'id': 6194}, {'name'...   
3      [{'name': 'Twentieth Century Fox Film Corporat...   
4      [{'name': 'Sandollar Productions', 'id': 5842}...   
...                                                  ...   
45460  [{'name': 'Westdeutscher Rundfunk (WDR)', 'id'...   
45462             [{'name': 'Sine Olivia', 'id': 19653}]   
45463  [{'name': 'American World Pictures', 'id': 6165}]   
45464               [{'name': 'Yermoliev', 'id': 88753}]   
45465                                                 []   

                                    production_countries release_date  \
0      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-10-30   
1      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-15   
2      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
3      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
4      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-02-10   
...                                                  ...          ...   
45460  [{'iso_3166_1': 'CA', 'name': 'Canada'}, {'iso...   1991-05-13   
45462      [{'iso_3166_1': 'PH', 'name': 'Philippines'}]   2011-11-17   
45463  [{'iso_3166_1': 'US', 'name': 'United States o...   2003-08-01   
45464           [{'iso_3166_1': 'RU', 'name': 'Russia'}]   1917-10-21   
45465   [{'iso_3166_1': 'GB', 'name'

Vamos a dejar solo el nombre en la columna production_companies

In [63]:
df_peliculas['production_companies']=df_peliculas['production_companies'].apply(ast.literal_eval)

In [64]:
df_peliculas['production_companies']=df_peliculas['production_companies'].apply(lambda x: [d['name']for d in x])

Vamos a dejar solo el nombre en la columna production_countries

In [65]:
df_peliculas['production_countries']=df_peliculas['production_countries'].apply(ast.literal_eval)

In [66]:
df_peliculas['production_countries']=df_peliculas['production_countries'].apply(lambda x: [d['name']for d in x])

In [67]:
df_peliculas

belongs_to_collection    budget  \
0      {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1                                                    NaN  65000000   
2      {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3                                                    NaN  16000000   
4      {'id': 96871, 'name': 'Father of the Bride Col...         0   
...                                                  ...       ...   
45460                                                NaN         0   
45462                                                NaN         0   
45463                                                NaN         0   
45464                                                NaN         0   
45465                                                NaN         0   

                             genres      id original_language  \
0       [Animation, Comedy, Family]     862                en   
1      [Adventure, Fantasy, Family]    8844                en   
2                 [Romance, Comedy]   15602                en   
3          [Comedy, Drama, Romance]   31357                en   
4                          [Comedy]   11862                en   
...                             ...     ...               ...   
45460      [Drama, Action, Romance]   30840                en   
45462                       [Drama]  111109                tl   
45463     [Action, Drama, Thriller]   67758                en   
45464                            []  227506                en   
45465                            []  461257                en   

                                                overview popularity  \
0      Led by Woody, Andy's toys live happily in his ...  21.946943   
1      When siblings Judy and Peter discover an encha...  17.015539   
2      A family wedding reignites the ancient feud be...    11.7129   
3      Cheated on, mistreated and stepped on, the wom...   3.859495   
4      Just when George Banks has recovered from his ...   8.387519   
...                                                  ...        ...   
45460  Yet another version of the classic epic, with ...   5.683753   
45462  An artist struggles to finish his work while a...   0.178241   
45463  When one of her hits goes wrong, a professiona...   0.903007   
45464  In a small town live two brothers, one a minis...   0.003503   
45465  50 years after decriminalisation of homosexual...   0.163015   

                                    production_companies  \
0                              [Pixar Animation Studios]   
1      [TriStar Pictures, Teitler Film, Interscope Co...   
2                         [Warner Bros., Lancaster Gate]   
3               [Twentieth Century Fox Film Corporation]   
4           [Sandollar Productions, Touchstone Pictures]   
...                                                  ...   
45460  [Westdeutscher Rundfunk (WDR), Working Title F...   
45462                                      [Sine Olivia]   
45463                          [American World Pictures]   
45464                                        [Yermoliev]   
45465                                                 []   

                                    production_countries release_date  \
0                             [United States of America]   1995-10-30   
1                             [United States of America]   1995-12-15   
2                             [United States of America]   1995-12-22   
3                             [United States of America]   1995-12-22   
4                             [United States of America]   1995-02-10   
...                                                  ...          ...   
45460  [Canada, Germany, United Kingdom, United State...   1991-05-13   
45462                                      [Philippines]   2011-11-17   
45463                         [United States of America]   2003-08-01   
45464                                           [Russia]   1917-10-21   
45465                               

In [68]:
df_peliculas.to_csv('peliculas.csv', index=False)

Vamos a crear otro dataset con los datos de credits

In [69]:
df_credits =pd.read_csv('..\Proyecto individual Henry\Data\credits (1).csv')


Vemos algunos datos del dataset

In [70]:
df_credits.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45476 entries, 0 to 45475
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   cast    45476 non-null  object
 1   crew    45476 non-null  object
 2   id      45476 non-null  int64 
dtypes: int64(1), object(2)
memory usage: 1.0+ MB


In [71]:
df_crew= df_credits.copy()

In [72]:
df_crew =df_crew.drop('cast', axis=1)

In [73]:
df_crew.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45476 entries, 0 to 45475
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   crew    45476 non-null  object
 1   id      45476 non-null  int64 
dtypes: int64(1), object(1)
memory usage: 710.7+ KB


Convierto la cadena de texto en lista de diccionario utilizando la función ast.literal_eval del modulo ast

In [74]:
import ast
df_crew['crew']= df_crew['crew'].apply(ast.literal_eval)

Creo una nueva columna en df_crew llamada job_crew donde cada elemento es una lista de valores de 'job'

In [75]:
df_crew['job_crew']= df_crew['crew'].apply(lambda x :[d['job'] for d in x])

Creo una nueva columna en df_crew llamada name_crew donde cada elemento es una lista de valores de 'name'



In [76]:
df_crew['name_crew']= df_crew['crew'].apply(lambda x :[d['name'] for d in x])

In [77]:
df_crew

crew      id  \
0      [{'credit_id': '52fe4284c3a36847f8024f49', 'de...     862   
1      [{'credit_id': '52fe44bfc3a36847f80a7cd1', 'de...    8844   
2      [{'credit_id': '52fe466a9251416c75077a89', 'de...   15602   
3      [{'credit_id': '52fe44779251416c91011acb', 'de...   31357   
4      [{'credit_id': '52fe44959251416c75039ed7', 'de...   11862   
...                                                  ...     ...   
45471  [{'credit_id': '5894a97d925141426c00818c', 'de...  439050   
45472  [{'credit_id': '52fe4af1c3a36847f81e9b15', 'de...  111109   
45473  [{'credit_id': '52fe4776c3a368484e0c8387', 'de...   67758   
45474  [{'credit_id': '533bccebc3a36844cf0011a7', 'de...  227506   
45475  [{'credit_id': '593e676c92514105b702e68e', 'de...  461257   

                                                job_crew  \
0      [Director, Screenplay, Screenplay, Screenplay,...   
1      [Executive Producer, Screenplay, Original Musi...   
2        [Director, Characters, Writer, Sound Recordist]   
3      [Director, Screenplay, Producer, Producer, Pro...   
4      [Original Music Composer, Director of Photogra...   
...                                                  ...   
45471  [Director, Producer, Camera Supervisor, Script...   
45472  [Director, Writer, Production Design, Music, E...   
45473  [Director, Screenplay, Screenplay, Original Mu...   
45474                               [Director, Producer]   
45475                                         [Director]   

                                               name_crew  
0      [John Lasseter, Joss Whedon, Andrew Stanton, J...  
1      [Larry J. Franco, Jonathan Hensleigh, James Ho...  
2      [Howard Deutch, Mark Steven Johnson, Mark Stev...  
3      [Forest Whitaker, Ronald Bass, Ronald Bass, Ez...  
4      [Alan Silvestri, Elliot Davis, Nancy Meyers, N...  
...                                                  ...  
45471  [Hamid Nematollah, Hamid Nematollah, Farshad M...  
45472  [Lav Diaz, Lav Diaz, Dante Perez, Lav Diaz, La...  
45473  [Mark L. Lester, C. Courtney Joyner, Jeffrey G...  
45474            [Yakov Protazanov, Joseph N. Ermolieff]  
45475                                    [Daisy Asquith]  

[45476 rows x 4 columns]

Elimino la columna crew del df_crew

In [78]:
df_crew = df_crew.drop('crew', axis=1)

In [79]:
df_crew

id                                           job_crew  \
0         862  [Director, Screenplay, Screenplay, Screenplay,...   
1        8844  [Executive Producer, Screenplay, Original Musi...   
2       15602    [Director, Characters, Writer, Sound Recordist]   
3       31357  [Director, Screenplay, Producer, Producer, Pro...   
4       11862  [Original Music Composer, Director of Photogra...   
...       ...                                                ...   
45471  439050  [Director, Producer, Camera Supervisor, Script...   
45472  111109  [Director, Writer, Production Design, Music, E...   
45473   67758  [Director, Screenplay, Screenplay, Original Mu...   
45474  227506                               [Director, Producer]   
45475  461257                                         [Director]   

                                               name_crew  
0      [John Lasseter, Joss Whedon, Andrew Stanton, J...  
1      [Larry J. Franco, Jonathan Hensleigh, James Ho...  
2      [Howard Deutch, Mark Steven Johnson, Mark Stev...  
3      [Forest Whitaker, Ronald Bass, Ronald Bass, Ez...  
4      [Alan Silvestri, Elliot Davis, Nancy Meyers, N...  
...                                                  ...  
45471  [Hamid Nematollah, Hamid Nematollah, Farshad M...  
45472  [Lav Diaz, Lav Diaz, Dante Perez, Lav Diaz, La...  
45473  [Mark L. Lester, C. Courtney Joyner, Jeffrey G...  
45474            [Yakov Protazanov, Joseph N. Ermolieff]  
45475                                    [Daisy Asquith]  

[45476 rows x 3 columns]

In [80]:
df_crew.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45476 entries, 0 to 45475
Data columns (total 3 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   id         45476 non-null  int64 
 1   job_crew   45476 non-null  object
 2   name_crew  45476 non-null  object
dtypes: int64(1), object(2)
memory usage: 1.0+ MB


Vemos los nulos

In [82]:
new_dupli= df_crew[df_crew.duplicated('id')]
print(new_dupli)

           id                                           job_crew  \
1465   105045                 [Director, Writer, Writer, Writer]   
9165     5511  [Director of Photography, Producer, Screenplay...   
9327    23305                         [Director, Writer, Writer]   
12066   14788  [Director, Director of Photography, Editor, Wr...   
13375  141971                                 [Director, Writer]   
15074   22649  [Director of Photography, Costume Design, Art ...   
15765   13209         [Director, Writer, Writer, Cinematography]   
16764  141971                                 [Director, Writer]   
20842   77221  [Screenplay, Original Music Composer, Director...   
20898  109962               [Director, Theatre Play, Screenplay]   
21115   84198                               [Director, Director]   
21164  119916                                 [Author, Director]   
21853  152795  [Producer, Casting, Novel, Producer, Music, Ci...   
22150   18440                                 [D

In [83]:
new_dupli.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 44 entries, 1465 to 45275
Data columns (total 3 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   id         44 non-null     int64 
 1   job_crew   44 non-null     object
 2   name_crew  44 non-null     object
dtypes: int64(1), object(2)
memory usage: 1.4+ KB


Se creara una lista de nombre en los cuales job coicida con la palabra Director

In [84]:
df_director = df_crew[df_crew['job_crew'].apply (lambda x : 'Director'in x)]

In [85]:
director_names = df_director['name_crew'].tolist()

In [86]:
director_names

[['John Lasseter',
  'Joss Whedon',
  'Andrew Stanton',
  'Joel Cohen',
  'Alec Sokolow',
  'Bonnie Arnold',
  'Ed Catmull',
  'Ralph Guggenheim',
  'Steve Jobs',
  'Lee Unkrich',
  'Ralph Eggleston',
  'Robert Gordon',
  'Mary Helen Leasman',
  'Kim Blanchette',
  'Marilyn McCoppen',
  'Randy Newman',
  'Dale E. Grahn',
  'Robin Cooper',
  'John Lasseter',
  'Pete Docter',
  'Joe Ranft',
  'Patsy Bouge',
  'Norm DeCarlo',
  'Ash Brannon',
  'Randy Newman',
  'Roman Figun',
  'Don Davis',
  'James Flamberg',
  'Mary Beth Smith',
  'Rick Mackay',
  'Susan Bradley',
  'William Reeves',
  'Randy Newman',
  'Andrew Stanton',
  'Pete Docter',
  'Gary Rydstrom',
  'Karen Robert Jackson',
  'Chris Montan',
  'Rich Quade',
  'Michael Berenstein',
  'Colin Brady',
  'Davey Crockett Feiten',
  'Angie Glocka',
  'Rex Grignon',
  'Tom K. Gurney',
  'Jimmy Hayward',
  'Hal T. Hickel',
  'Karen Kiser',
  'Anthony B. LaMolinara',
  'Guionne Leroy',
  'Bud Luckey',
  'Les Major',
  'Glenn McQueen',
  

Queremos saber cuantos directores son para lo cual realizamos lo siguente

In [87]:
numero_director = len(director_names)

In [89]:
numero_director

44589

Luego de haber realizado ETL vamos a proceder a relizar el Analisis Exploratorio de los datos (EDA)

Teniendo en cuenta que lo que queremos obtener es un sistema de recomendación de peliculas vamos a 
empezar con el EDA


########Queremos saber cuales son las palabras mas utilizadas en los titulos por ejemplo

La información que vamos a analizar es un csv de peliculas llamado movies.csv

Vamos a proceder con la lectura del dataset. Para lo cual necesitamos esta librerias. En este momento no importamos librerias debido a que ya se hizo al comenzar este archivo.

In [ ]:
#import pandas as pd 
#import matplotlib.pyplot as pyplot 
#import seaborn as sns


Se creo un dataset del archivo movies.csv. Que lo hicimos en nuestro ETL. Por lo cual no lo vamos a hacer ahora.

In [ ]:
#df_peliculas = pd.read_csv ('..\Proyecto individual Henry\Data\movies_dataset (1).csv')

Vamos a ver las caracteristicas principales. Nuestro dataset contaba originalmente con registros 45466 y
24 columnas


Cuando se llevo a cabo el ETL se elimino las siguientes columnas: 'video', 'imdb_id', 'adult', 'original_title', 'poster_path' y 'homepage'. Es decir que se eliminaron 6 columnas y tambien se  crearon 3 nuevas columnas 'realease_year','genres_genres' y 'return'.Se eliminaron 120 filas en la limpieza. Quedando con 45346 filas y 21 columnas.

In [170]:
df_peliculas

belongs_to_collection    budget  \
0      {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1                                                    NaN  65000000   
2      {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3                                                    NaN  16000000   
4      {'id': 96871, 'name': 'Father of the Bride Col...         0   
...                                                  ...       ...   
45460                                                NaN         0   
45462                                                NaN         0   
45463                                                NaN         0   
45464                                                NaN         0   
45465                                                NaN         0   

                                                  genres      id  \
0      [{'id': 16, 'name': 'Animation'}, {'id': 35, '...     862   
1      [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...    8844   
2      [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   15602   
3      [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   31357   
4                         [{'id': 35, 'name': 'Comedy'}]   11862   
...                                                  ...     ...   
45460  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...   30840   
45462                      [{'id': 18, 'name': 'Drama'}]  111109   
45463  [{'id': 28, 'name': 'Action'}, {'id': 18, 'nam...   67758   
45464                                                 []  227506   
45465                                                 []  461257   

      original_language                                           overview  \
0                    en  Led by Woody, Andy's toys live happily in his ...   
1                    en  When siblings Judy and Peter discover an encha...   
2                    en  A family wedding reignites the ancient feud be...   
3                    en  Cheated on, mistreated and stepped on, the wom...   
4                    en  Just when George Banks has recovered from his ...   
...                 ...                                                ...   
45460                en  Yet another version of the classic epic, with ...   
45462                tl  An artist struggles to finish his work while a...   
45463                en  When one of her hits goes wrong, a professiona...   
45464                en  In a small town live two brothers, one a minis...   
45465                en  50 years after decriminalisation of homosexual...   

      popularity                               production_companies  \
0      21.946943     [{'name': 'Pixar Animation Studios', 'id': 3}]   
1      17.015539  [{'name': 'TriStar Pictures', 'id': 559}, {'na...   
2        11.7129  [{'name': 'Warner Bros.', 'id': 6194}, {'name'...   
3       3.859495  [{'name': 'Twentieth Century Fox Film Corporat...   
4       8.387519  [{'name': 'Sandollar Productions', 'id': 5842}...   
...          ...                                                ...   
45460   5.683753  [{'name': 'Westdeutscher Rundfunk (WDR)', 'id'...   
45462   0.178241             [{'name': 'Sine Olivia', 'id': 19653}]   
45463   0.903007  [{'name': 'American World Pictures', 'id': 6165}]   
45464   0.003503               [{'name': 'Yermoliev', 'id': 88753}]   
45465   0.163015                                                 []   

                                    production_countries release_date  ...  \
0      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-10-30  ...   
1      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-15  ...   
2      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22  ...   
3      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22  ...   
4      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-02-10  ...   
...                                                  ...          ...  ...   
45460  [{'iso_3166_1': 'CA', 'name': 'Can

In [89]:
df_peliculas.to_csv('peliculas.csv', index=False)

Procedemos a analizar cada una de las columnas

In [90]:
df_peliculas.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 45346 entries, 0 to 45465
Data columns (total 21 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   belongs_to_collection  4485 non-null   object        
 1   budget                 45346 non-null  int64         
 2   genres                 45346 non-null  object        
 3   id                     45346 non-null  object        
 4   original_language      45346 non-null  object        
 5   overview               44405 non-null  object        
 6   popularity             45346 non-null  object        
 7   production_companies   45346 non-null  object        
 8   production_countries   45346 non-null  object        
 9   release_date           45346 non-null  datetime64[ns]
 10  revenue                45346 non-null  float64       
 11  runtime                45100 non-null  float64       
 12  spoken_languages       45346 non-null  object        
 13  s

Procedemos a leer los datos

Voy a realizar un grafico de conteo

In [91]:
df_peliculas['original_language']= df_peliculas['original_language'].astype(str)
df_peliculas['genres_genres']= df_peliculas['genres_genres'].astype(str)


In [92]:
df_peliculas['genres_genres']

0         ['Animation', 'Comedy', 'Family']
1        ['Adventure', 'Fantasy', 'Family']
2                     ['Romance', 'Comedy']
3            ['Comedy', 'Drama', 'Romance']
4                                ['Comedy']
                        ...                
45460        ['Drama', 'Action', 'Romance']
45462                             ['Drama']
45463       ['Action', 'Drama', 'Thriller']
45464                                    []
45465                                    []
Name: genres_genres, Length: 45346, dtype: object